File: Examples/CSharp/FormattingAndManipulatingOCR/SettingtheOcrEngineAutomaticallyDetectTextBlocks.cs
using System.IO;
using Aspose.OCR;
using System;

namespace Aspose.OCR.Examples.CSharp.FormattingAndManipulatingOCR
{
    public class SettingtheOcrEngineAutomaticallyDetectTextBlocks
    {
        public static void Run()
        {
            // ExStart:SettingtheOcrEngineAutomaticallyDetectTextBlocks
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_OMR();

            // Create an instance of OcrEngine class
            OcrEngine ocr = new OcrEngine();

            // Set the DetectTextRegions to true
            ocr.Config.DetectTextRegions = true;

            // Set the Image property of OcrEngine by reading an image file
            ocr.Image = ImageStream.FromFile(dataDir + "Sampleocr.bmp");

            // Perform OCR operation on the image
            if (ocr.Process())
            {
                // Display results
                Console.WriteLine(ocr.Text);
            }
            // ExEnd:SettingtheOcrEngineAutomaticallyDetectTextBlocks
        }
    }
}

File: Examples/VisualBasic/FormattingAndManipulatingOCR/SettingtheOcrEngineAutomaticallyDetectTextBlocks.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.OCR
Imports System

Namespace Aspose.OCR.Examples.VisualBasic.FormattingAndManipulatingOCR
    Public Class SettingtheOcrEngineAutomaticallyDetectTextBlocks
        Public Shared Sub Run()
            ' ExStart:SettingtheOcrEngineAutomaticallyDetectTextBlocks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_OCR()

            'Create an instance of OcrEngine class
            Dim ocr As New OcrEngine()

            'Set the DetectTextRegions to true
            ocr.Config.DetectTextRegions = True

            'Set the Image property of OcrEngine by reading an image file
            ocr.Image = ImageStream.FromFile(dataDir & "Sampleocr.bmp")

            'Perform OCR operation on the image
            If ocr.Process() Then
                'Display results
                Console.WriteLine(ocr.Text)
            End If
        End Sub
        ' ExEnd:SettingtheOcrEngineAutomaticallyDetectTextBlocks
    End Class
End Namespace
